src/flowcean/transforms/one_cold.py

```diff
@@ -0,0 +1,153 @@
+                to see if all values belong to a category. If an unknown value
+                is found which does not belong to any category, a
+                NoMatchingCategoryError is thrown. To perform this check, the
+                dataframe must be materialised, resulting in a potential
+                performance decrease. Therefore it defaults to false.
+        """
+        self.feature_category_mapping = {
+            feature: {f"{feature}_{value}": value for value in values}
+            for feature, values in feature_categories.items()
+        }
+        self.check_for_missing_categories = check_for_missing_categories
+
+    @override
+    def apply(
+        self,
+        data: pl.DataFrame,
+    ) -> pl.DataFrame:
+        """Transform data with this one hot transformation.
+
+        Transform data with this one hot transformation and return the
+        resulting dataframe.
+
+        Args:
+            data: The data to transform.
+
+        Returns:
+            The transformed data.
+        """
+        if len(self.feature_category_mapping) == 0:
+            raise NoCategoriesError
+        for (
+            feature,
+            category_mappings,
+        ) in self.feature_category_mapping.items():
+            data = data.with_columns(
+                [
+                    pl.col(feature).ne(value).cast(pl.Int64).alias(name)
+                    for name, value in category_mappings.items()
+                ]
+            ).drop(feature)
+
+            if self.check_for_missing_categories and (
+                not data.select(
+                    [
+                        pl.col(name).cast(pl.Boolean)
+                        for name in category_mappings
+                    ]
+                )
+                .select(pl.all_horizontal(pl.all()).all())
+                .item(0, 0)
+            ):
+                raise NoMatchingCategoryError
+        return data
+
+    @classmethod
+    def from_dataframe(
+        cls,
+        data: pl.DataFrame,
+        features: Iterable[str],
+        *,
+        check_for_missing_categories: bool = False,
+    ) -> Self:
+        """Creates a new one-hot transformation based on sample data.
+
+        Args:
+            data: A dataframe containing sample data for determining the
+                categories of the transform.
+            features: Name of the features for which the one hot transformation
+                will determine the categories.
+            check_for_missing_categories: If set to true, a check is performed
+                to see if all values belong to a category. If an unknown value
+                is found which does not belong to any category, a
+                NoMatchingCategoryError is thrown. To perform this check, the
+                dataframe must be materialised, resulting in a potential
+                performance decrease. Therefore it defaults to false.
+        """
+        # Derive categories from the data frame
```

Comment:
If we decide not to put inline comments, we should probably remove this one as well.

Reply:
Do you mean the flag or the comment? I think it's still relevant for the user to have the choice - they don't need to understand how it works under the hood as long as it does what they want :D

Reply:
I just meant the comment ;)